packages/core/src/main/conversation/ConversationService.ts

```diff
@@ -886,17 +937,47 @@ class ConversationService {
     };
   }
 
+  public createText(text: string, messageId?: string): PayloadBundleOutgoingUnsent;
   public createText(
     text: string,
     linkPreviews?: LinkPreviewUploadedContent[],
-    messageId: string = ConversationService.createId()
+    messageId?: string
+  ): PayloadBundleOutgoingUnsent;
+  public createText(text: string, mentions?: MentionContent[], messageId?: string): PayloadBundleOutgoingUnsent;
+  public createText(
+    text: string,
+    linkPreviews?: LinkPreviewUploadedContent[],
+    mentions?: MentionContent[],
+    messageId?: string
+  ): PayloadBundleOutgoingUnsent;
+  public createText(
```

Comment:
The list of parameters here is getting longer and longer. Let's refactor it applying the [Builder pattern](https://github.com/torokmark/design_patterns_in_typescript/tree/master/builder).

**Idea:**

`.createText(text).withLinkPreviews(...).withMentions(...)`

Reply:
https://github.com/wireapp/wire-web-packages/pull/1099